File: Add_two_numbers_represented_by_linked_lists.c
#include <stdio.h>
#include <stdlib.h>

struct Node{
    int data;
    struct Node* next;
};

struct Node* create_node(int val){
    struct Node* newnode=malloc(sizeof(struct Node));
    newnode->data=val;
    newnode->next=NULL;
    return newnode;
}

struct Node* add_lists(struct Node* l1, struct Node* l2){
    struct Node *res=NULL, *tail=NULL;
    int carry=0;
    while(l1 || l2 || carry){
        int sum=(l1?l1->data:0)+(l2?l2->data:0)+carry;
        carry=sum/10;
        struct Node* newnode=create_node(sum%10);
        if(!res)res=tail=newnode;
        else{tail->next=newnode; tail=newnode;}
        if(l1)l1=l1->next;
        if(l2)l2=l2->next;
    }
    return res;
}

void print_list(struct Node* head){
    while(head){printf("%d ",head->data); head=head->next;}
    printf("\n");
}

int main(){
    struct Node *l1=create_node(2); l1->next=create_node(4); l1->next->next=create_node(3);
    struct Node *l2=create_node(5); l2->next=create_node(6); l2->next->next=create_node(4);
    struct Node *res=add_lists(l1,l2);
    print_list(res);
    return 0;
}

File: Add_two_numbers_represented_by_linked_lists.py
class Node:
    def __init__(self,data):
        self.data=data
        self.next=None

def add_lists(l1,l2):
    dummy=Node(0)
    tail=dummy
    carry=0
    while l1 or l2 or carry:
        s=(l1.data if l1 else 0)+(l2.data if l2 else 0)+carry
        carry=s//10
        tail.next=Node(s%10)
        tail=tail.next
        l1=l1.next if l1 else None
        l2=l2.next if l2 else None
    return dummy.next

def print_list(head):
    while head:
        print(head.data,end=" ")
        head=head.next
    print()

l1=Node(2); l1.next=Node(4); l1.next.next=Node(3)
l2=Node(5); l2.next=Node(6); l2.next.next=Node(4)
res=add_lists(l1,l2)
print_list(res)
